### PR TITLE
Upgrade to Sync 5.0.21

### PR DESCRIPTION
**Pull request by cmelchior**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
+### Enhancements
+* None.
+
+### Fixes
+* In some cases a frozen Realm of the wrong version could be returned.
+* Upgrading files with string primary keys would result in a file where it was not possible to find the objects by primary key. ([Core issue #3893](https://github.com/realm/realm-core/pull/3893), since 7.0.0)
+
+### Compatibility
+* Realm Object Server: 3.23.1 or later.
+* File format: Generates Realms with format v10 (Reads and upgrades all previous formats from Realm Java 2.0 and later).
```
Comment (rorbech) [inline CHANGELOG.md:12]: How does this play with the below comment about "Fileformat has been bumped from 10 to 11."

**Review by rorbech — APPROVED** [review]
Looks good. Only minor comment on version numbers in change log.

File: CHANGELOG.md (hunk 1 above)
Comment (cmelchior) [reply CHANGELOG.md:12]: `2.0` is the release number which isn't associated with the file format number in any way.

Comment (rorbech) [reply CHANGELOG.md:12]: It wasn't the `2.0` but the `File format: Generates Realms with format v10` I was questioning.

**Pull request merged**